aiidalab/utils.py

```diff
@@ -202,6 +203,31 @@ def get_package_by_name(packages: dict[str, Package], name: str) -> Package | No
     return None
 
 
+def is_valid_version(version: str) -> bool:
+    """Return True if given string is a PEP440-compliant version, False otherwise."""
+    try:
+        parse(version)
+    except InvalidVersion:
+        return False
+    else:
+        return True
+
+
+def sort_semantic(versions: Iterable[str], prereleases: bool = False) -> list[str]:
+    """Sort app versions semantically, latest first.
+    Optionally filter out the pre-release versions.
+
+    :raises: packaging.version.InvalidVersion if the input list contains invalid version.
+    """
+    if versions is None:
```

Comment:
If the versions parameter can accept `None` as a valid value, the type annotation should explicitly reflect this.

Reply:
Thanks for catching this. I was too cautious here, I decided to simply remove the check, this function expects an iterable. I tested that passing an empty iterable works and returns an empty list.